crates/api/src/chronicle_graphql/mod.rs

```diff
@@ -458,6 +523,73 @@ impl async_graphql::extensions::ExtensionFactory for AuthFromJwt {
     }
 }
 
+pub struct OpaCheck;
+
+#[async_trait::async_trait]
+impl async_graphql::extensions::Extension for OpaCheck {
+    #[instrument(level = "debug", skip_all, ret(Debug))]
+    async fn resolve(
+        &self,
+        ctx: &async_graphql::extensions::ExtensionContext<'_>,
+        info: async_graphql::extensions::ResolveInfo<'_>,
+        next: async_graphql::extensions::NextResolve<'_>,
+    ) -> async_graphql::ServerResult<Option<async_graphql::Value>> {
+        use async_graphql::ServerError;
+        use serde_json::Value;
+        if let (Some(identity), Some(opa_executor)) = (
+            ctx.data_opt::<AuthId>(),
+            ctx.data_opt::<Arc<Mutex<WasmtimeOpaExecutor>>>(),
+        ) {
+            let opa_context = Value::Object(
```

Comment:
We should probably get the operation type in here too, Subscribe / Query / Mutate